src/pybind/mgr/telemetry/module.py

```diff
@@ -390,9 +395,9 @@ def gather_device_report(self):
                                                            host, anon_host))
 
             # anonymize the smartctl report itself
-            for k in ['serial_number']:
-                if k in m:
-                    m.pop(k)
+            serial = devid.rsplit('_', 1)[1]
+            m_str = json.dumps(m)
+            m = json.loads(m_str.replace(serial, 'deleted'))
```

Comment:
separate this into a separate git commit?

Reply:
separated